cli/admin/src/main/java/io/pravega/cli/admin/AdminCommand.java

```diff
@@ -390,11 +390,21 @@ public SegmentHelper instantiateSegmentHelper(CuratorFramework zkClient) {
                 .containerCount(getServiceConfig().getContainerCount())
                 .build();
         HostControllerStore hostStore = HostStoreFactory.createStore(hostMonitorConfig, StoreClientFactory.createZKStoreClient(zkClient));
-        ClientConfig clientConfig = ClientConfig.builder()
-                .controllerURI(URI.create(getCLIControllerConfig().getControllerGrpcURI()))
-                .validateHostName(getCLIControllerConfig().isAuthEnabled())
-                .credentials(new DefaultCredentials(getCLIControllerConfig().getPassword(), getCLIControllerConfig().getUserName()))
-                .build();
+
+        ClientConfig.ClientConfigBuilder clientConfigBuilder = ClientConfig.builder()
+                .controllerURI(URI.create(getCLIControllerConfig().getControllerGrpcURI()));
+
+        if (getCLIControllerConfig().isAuthEnabled()) {
+            clientConfigBuilder.credentials(new DefaultCredentials(getCLIControllerConfig().getPassword(),
+                    getCLIControllerConfig().getUserName()));
+        }
+        if (getCLIControllerConfig().isTlsEnabled()) {
+            clientConfigBuilder.trustStore(getCLIControllerConfig().getTruststore())
+                    .validateHostName(false);
```

Comment:
Why is this hardcoded? Should this be yet another parameter?

Reply:
This value wont be changed. For example in admin/utils/TestUtils.java